addons/website/static/src/snippets/s_image_gallery/000.js

```diff
@@ -26,7 +26,7 @@ const GalleryWidget = publicWidget.Widget.extend({
      * @param {Event} ev
      */
     _onClickImg: function (ev) {
-        if (this.$modal) {
+        if (this.$modal || ev.currentTarget.parentElement.matches("a")) {
```

Comment:
Fine like this :+1:  But to avoid going to the parent to check, you could also do 
```js
ev.currentTarget.matches("a img")
```
since it is a selector :)